skbio/sequence/_iupac_sequence.py

```diff
@@ -164,23 +167,71 @@ def _motifs(self):
 
     @overrides(Sequence)
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 validate=True, case_insensitive=False):
+                 validate=True, lowercase=False):
         super(IUPACSequence, self).__init__(
             sequence, metadata, positional_metadata)
 
-        if case_insensitive:
-            self._convert_to_uppercase()
+        # Note we are checking identity against the False object here and not
+        # just falsiness
+        if lowercase is not False:
+            lowercase_mask = self._bytes > self._ascii_lowercase_boundary
+            self._convert_to_uppercase(lowercase_mask)
+            if not isinstance(lowercase, bool):
```

Comment:
I tried it and `None` is set as a valid column in `positional_metadata`. I believe @jairideout and I talked about a similar issue a while back having to do with situations like `lowercase=[]` and `lowercase=()`. We decided to go the route of letting pandas handle anything that's not a `bool`. Honestly I think we may want to constrain it to `bool` or `six.string_types`, but not sure.